src/cascadia/WindowsTerminal/BaseWindow.h

```diff
@@ -115,30 +114,59 @@ class BaseWindow
                 SWP_NOZORDER | SWP_NOACTIVATE);
 
             _currentDpi = uDpi;
-            NewScale(uDpi);
         }
         _inDpiChange = false;
         return 0;
     }
 
-    virtual void NewScale(UINT dpi) = 0;
-
     virtual void OnResize(const UINT width, const UINT height) = 0;
     virtual void OnMinimize() = 0;
     virtual void OnRestore() = 0;
 
-    RECT GetWindowRect() const
+    RECT GetWindowRect() const noexcept
     {
         RECT rc = { 0 };
         ::GetWindowRect(_window, &rc);
         return rc;
     }
 
-    HWND GetHandle() noexcept
+    HWND GetHandle() const noexcept
     {
         return _window;
     };
 
+    float GetCurrentDpiScale() const noexcept
+    {
+        const auto dpi = ::GetDpiForWindow(_window);
+        const auto scale = float(dpi) / float(USER_DEFAULT_SCREEN_DPI);
+        return scale;
+    }
+
+    //// Gets the physical size of the client area of the HWND in _window
+    SIZE GetPhysicalSize() const noexcept
+    {
+        RECT rect = {};
+        GetClientRect(_window, &rect);
+        const auto windowsWidth = rect.right - rect.left;
+        const auto windowsHeight = rect.bottom - rect.top;
+        return SIZE{ windowsWidth, windowsHeight };
+    }
+
+    //// Gets the logical size of physical size specified by the parameter physicalSize

```

Comment:
I was kinda hoping that the comments would be a little more descriptive - what's the difference between physical size and logical size? What's logical size used for?

Reply:
I will add more and link to https://docs.microsoft.com/en-us/windows/desktop/LearnWin32/dpi-and-device-independent-pixels